### ISSUE TITLE
Streaming LDIF to file has stopped working in 2.8.1

### ISSUE THREAD

**Issue by holgeric** [cannatag/ldap3#879]
The following example from https://ldap3.readthedocs.io/en/latest/ldif.html works fine with ldap3 v2.8, but gives an error with 2.8.1.

This on Windows 10 (1909) and with Python 3.8.5.

```
from ldap3 import Connection, LDIF

c = Connection(None, client_strategy=LDIF)
c.stream = open('output.ldif', 'w')
with c:
    c.delete('cn=test1,o=test')
    c.delete('cn=test2,o=test')
```

Version 2.8:

```
C:\dev\tools\py>py -m pip install ldap3==2.8
Collecting ldap3==2.8
...
Successfully installed ldap3-2.8

C:\dev\tools\py>py testLdif.py
C:\dev\tools\py>type output.ldif
version: 1

dn: cn=test1,o=test
changetype: delete

dn: cn=test2,o=test
changetype: delete
```

Version 2.8.1:

```
C:\dev\tools\py>py -m pip install --upgrade ldap3==2.8.1
...
Successfully installed ldap3-2.8.1

C:\dev\tools\py>py testLdif.py
Traceback (most recent call last):
  File "testLdif.py", line 5, in <module>
    with c:
  File "C:\Users\...\Python38\lib\site-packages\ldap3\core\connection.py", line 534, in __enter__
    raise LDAPBindError('unable to bind')
ldap3.core.exceptions.LDAPBindError: unable to bind

```

**Comment by cannatag** [comment]
Hi, sorry for the long delay. Can you try the code in dev? It should work now.

**Comment by holgeric** [comment]
Hi, I tested with the latest in dev and it seems to work fine now.

**Comment by cannatag** [comment]
thanks. fixed in ldap3 v2.9
